### DISCUSSION TITLE
React Native SDK on iOS not work

### DISCUSSION THREAD

**Discussion by andybui2025** [ant-media/Ant-Media-Server#6502] [FAQs]
When I run the example code, Android runs perfectly, but iOS says could not sender when calling adapter.publish and "Ice candidate is added to list because remote description is not set yet" when calling adapter.play.
package json:  "react-native": "0.74.3", "@antmedia/react-native-ant-media": "1.0.8", "react-native-webrtc": "124.0.3".
Please help me

**Comment by zerostack0** [comment]


**Reply by andybui2025** [reply]
@mustafaboleken, 
- I had try swich to VP8 encoder and also not working.
- I am using the example https://github.com/ant-media/WebRTC-React-Native-SDK/tree/main/example
- I also tried switching to the react native webrtc version as in the example, but it only works on Android, iPhone does not work.
- Below log snippet code when i run adapter.publish(streamNameRef.current);
- 
![Screenshot 2024-07-18 at 10 12 42](https://github.com/user-attachments/assets/56769216-8684-4f61-85a8-af4c0b49c6cc)

**Comment by andybui2025** [comment]
Hi Ant-Media/Ant-Media-Server support,
Thanks for the good news for me. I hope the problem will resolve as soon as
possible./
Best regards,


On Tue, Jul 23, 2024 at 3:38 PM Mustafa BOLEKEN ***@***.***>
wrote:

> Hi @ThoaiBuiMCV <https://github.com/ThoaiBuiMCV>
>
> Thanks for the logs. I'm converting this discussion into issue. We will
> schedule it during sprint meetings.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/ant-media/discussions/6502#discussioncomment-10123725>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BBO5EZEACL2CIFMWTITOHIDZNYI6XAVCNFSM6AAAAABK7655PSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMJSGM3TENI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> .com>
>


-- 

Bùi Thanh Thoại

Mobile Developer

Phone: 0909995896

Email: ***@***.*** ***@***.***>

Website: www.mcv.com.vn

VPĐD: MCV Complex, Lô B7, đường 19A, KCX Tân Thuận, P.Tân Thuận Đông, Quận
7, TP. Hồ Chí Minh

**Reply by mekya** [reply]
Hi @ThoaiBuiMCV ,
We've just upgraded the react-native(0.75) and react-native-webrtc(124) versions, examples and checked them they are working both in Android and iOS.
https://github.com/ant-media/WebRTC-React-Native-SDK/pull/25

New version is available -> https://www.npmjs.com/package/@antmedia/react-native-ant-media

Let us know if you need help. We can make a live demo for Android and iOS

Cheers
Oguz
